### DISCUSSION TITLE
extend onSubmit to press of button for useSearchField

### DISCUSSION THREAD

**Discussion by PixeledCode** [adobe/react-spectrum#3689] [Q&A] [answered]
I was wondering what would be the most efficient way to extend the ability to trigger the onSubmit callback when using a submit button as well. 
Right now, I can think of creating another callback for just the button, but if possible, I would like to prevent that. 

Thanks again for this fantastic library.

**Comment by snowystinger** [comment, marked as answer]
I'm not quite sure what you're asking for. onSubmit is a form event. https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event so you wouldn't put it on a button. You'd put it on a form and you'd have a submit button inside that form.
Let me know if that answers your question.

**Reply by PixeledCode** [reply]
Hi, thanks for the answer. I was talking about the event handlers useSearchField provides https://react-spectrum.adobe.com/react-aria/useSearchField.html#events.

It neatly provides the text on the `Enter` click while in the search field. I thought if we could also call that event while clicking on a submit button

<img width="550" alt="Screenshot 2022-10-28 at 12 12 18 PM" src="https://user-images.githubusercontent.com/22215209/198520615-98a15cbe-60a0-4fd0-9e42-a83fa927fce9.png">

When used as a standalone search bar, I am not using `form`

**Reply by PixeledCode** [reply]
okay, so just found a solution that I think will work

  ```
<Button
   className="search-submit"
   type="submit"
   onPress={() => props.onSubmit(ref.current.value)     // <-- ref of input field
   iconOnly
   icon={<SearchIcon />}
>
   submit search
</Button>
```

**Reply by snowystinger** [reply]
I mean, you could also do something more like this so you don't need to track the ref
https://codesandbox.io/s/modest-kilby-wx6ck9?file=/src/App.js

**Reply by PixeledCode** [reply]
Yes that's better way to do it but it might not work for uncontrolled search field right?

**Reply by snowystinger** [reply]
It'd work for uncontrolled as well, I updated the codesandbox for uncontrolled.

**Reply by PixeledCode** [reply]
Yes that would work perfectly fine as well. Thanks for helping out!
